cogito/put.go

```diff
@@ -26,91 +45,167 @@ import (
 // Additionally, the script may emit metadata as a list of key-value pairs. This data is
 // intended for public consumption and will make it upstream, intended to be shown on the
 // build's page.
-func Put(log hclog.Logger, in io.Reader, out io.Writer, args []string) error {
-	var pi PutInput
+func Put(log hclog.Logger, in io.Reader, out io.Writer, args []string, putter Putter) error {
+	if err := putter.LoadConfiguration(in, args); err != nil {
+		return fmt.Errorf("put: %s", err)
```

Comment:
Does it make sense here and below adding the caller in the prefix? Example `put LoadConfiguration:...`

Reply:
Ah, I see the reasoning. It is indeed always confusing to decide the "point of view" when reporting errors. This is what I attempt to do in cogito:

The "LoadConfiguration" prefix (or anyway something clear enough to locate the error) should be added by the LoadConfiguration() function itself when returning an error. Example:

```
func LoadConfiguration() {
    ...
    return fmt.Errorf("parsing stdin: %s", ...)
}
```
where in this case the full error would become:
```
put: parsing stdin: ...
```
and be clear enough.

Reply:
Ah sure, I missed the "point of view".